qiskit_ibm/runtime/utils.py

```diff
@@ -108,6 +114,73 @@ def deserialize_from_settings(mod_name: str, class_name: str, settings: Dict) ->
     raise ValueError(f"Unable to find class {class_name} in module {mod_name}")
 
 
+def _set_int_keys_flag(obj: Union[Dict, List]) -> Union[Dict, List]:
+    """Recursively sets '__int_keys__' flag if dictionary uses integer keys
+
+    Args:
+        obj: dictionary or list
+
+    Returns:
+        obj with the '__int_keys__' flag set if dictionary uses integer key
+    """
+    if isinstance(obj, dict):
+        for k, v in list(obj.items()):
```

Comment:
It is needed since we are adding items to dict while looping through it. Otherwise I get this error:
```
RuntimeError: dictionary changed size during iteration
```